crates/acvm/src/lib.rs

```diff
@@ -27,7 +27,30 @@ pub trait PartialWitnessGenerator {
         initial_witness: &mut BTreeMap<Witness, FieldElement>,
         gates: Vec<Gate>,
     ) -> Result<(), OPCODE>;
+
+    fn get_value(
```

Comment:
Why can't we get the value of a Witness instead of needing it to be an arithmetic?

Reply:
To avoid creating a witness, where we can use an arithmetic expression instead.

Reply:
If its a constant, shouldn't it be a FieldElement and not a witness?

Reply:
Yes it is, this is the point to use an Arithmetic inside a Directive instead of a witness.
An arithmetic expression can be a field element, or a witness (or a linear combination of witness, or a scalar product of witness vectors).
When we have a constant, the Arithmetic is just a field element.